tests/testthat/test_pipeop_subsample.R

```diff
@@ -73,7 +73,40 @@ test_that("PipeOpSubsample works stratified", {
     table(list(Species = rep(c("setosa", "versicolor", "virginica"), 100))))
 })
 
+test_that("PipeOpSubsample works for grouped data", {
+  op = PipeOpSubsample$new()
+
+  test_df = data.frame(
+    target = runif(3000),
+    x1 = runif(3000),
+    x2 = runif(3000),
+    grp = sample(paste0("g", 1:100), 3000, replace = TRUE)
+  )
+  task = TaskRegr$new(id = "test", backend = test_df, target = "target")
+  task$set_col_roles("grp", "group")
+
+  train_out = op$train(list(task))[[1L]]
+  # grouped data are kept together
+  grps_out = table(train_out$groups$group)
+  expect_equal(
+    grps_out,
+    table(task$groups$group)[names(grps_out)]
+  )
+
+  op$param_set$set_values(frac = 2, replace = TRUE)
+  train_out = op$train(list(task))[[1L]]
+  # grouped data are kept together
+  grps_out = table(train_out$groups$group)
+  expect_equal(
+    grps_out,
+    table(task$groups$group)[names(grps_out)]
+  )
+
```

Comment:
test that fraction is adhered to as well as possible; even with imbalanced group sizes (e.g. one very large group).